zng/resolver/marshal.go

```diff
@@ -55,18 +58,25 @@ func fieldName(f reflect.StructField) string {
 	return f.Name
 }
 
+func encodeMarshaler(zctx *Context, b *zcode.Builder, v reflect.Value) (zng.Type, error) {
+	m, ok := v.Interface().(Marshaler)
+	if !ok {
+		return nil, fmt.Errorf("couldn't marshal interface: %v", v)
+	}
+	return m.MarshalZNG(zctx, b)
+}
+
 func encodeAny(zctx *Context, b *zcode.Builder, v reflect.Value) (zng.Type, error) {
+	if v.Type().Implements(marshalerType) {
+		return encodeMarshaler(zctx, b, v)
```

Comment:
Maybe simplify, getting rid of encodeMarhsaler and its "couldn't marshal interface" error?
```suggestion
		return v.Interface().(Marshaler).MarshalZNG(zctx, b)
```